commercial/app/controllers/BookOffersController.scala

```diff
@@ -20,46 +17,35 @@ class BookOffersController(bookFinder: BookFinder, bestsellersAgent: Bestsellers
   with implicits.Collections
   with implicits.Requests {
 
-  private def booksSample(isbns: Seq[String], segment: Segment): Future[Seq[Book]] =
-    bestsellersAgent.getSpecificBooks(isbns) map { specificBooks =>
-      (specificBooks ++ bestsellersAgent.bestsellersTargetedAt(segment)).distinctBy(_.isbn).take(4)
-    }
+  private def booksSample(isbns: Seq[String], segment: Segment): Seq[Book] =
+    (bestsellersAgent.getSpecificBooks(isbns) ++ bestsellersAgent.bestsellersTargetedAt(segment)).distinctBy(_.isbn).take(4)
+
+  private def isValidIsbn(isbn: String): Boolean = (isbn forall (_.isDigit)) && (isbn.length == 10 || isbn.length == 13)
```

Comment:
Ae all those parenthesis necessary? 
Also I would stick with the dot notation to call methods and not mix dot and space in the same statement. ie: `isbn.foreall(isDigit) && (isbn.length == 10 || isbn.length == 13)`